packages/amplify-appsync-simulator/src/velocity/value-mapper/integer.ts

```diff
@@ -0,0 +1,23 @@
+export class JavaInteger {
+  value: number;
+
+  constructor(val: number) {
+    this.value = Math.trunc(val);
+  }
+
+  valueOf(): number {
+    return this.value;
+  }
+
+  parseInt(val: any, radix: any = 10): JavaInteger {
+    return new JavaInteger(Number.parseInt(val, +radix));
```

Comment:
Why +radix , instead of radix.

Reply:
It will ensure that `radix` is converted to a number before passing it to `Number.parseInt()`. If `radix` happens to be a `JavaInteger` object instead of a normal JavaScript value, it will call `valueOf()` to convert it.